code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -104,4 +104,14 @@
 	melee_damage_lower = 15
 	melee_damage_upper = 18
 
+/mob/living/simple_animal/hostile/carp/cayenne/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(.)
+		regen_cooldown = world.time + REGENERATION_DELAY
+
+/mob/living/simple_animal/hostile/carp/cayenne/Life()
+	. = ..()
+	if(regen_cooldown < world.time)
+		heal_overall_damage(8)
+
```

Comment:
Move this procedure to the base space carp type and add a new variable to toggle the health regeneration for Cayenne and mega carps.